### PR TITLE
Add ViltForTokenClassification e.g. for Named-Entity-Recognition (NER)

### PR DESCRIPTION
**Pull request by gilad19**
# What does this PR do?

Adding ViltForTokenClassification in order to be able to fine-tune ViLT for a token classification task (e.g. as Named-Entity-Recognition). This allows leveraging both image and text for token classification tasks using ViLT. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

## Who can review?

@xhluca , @LysandreJik, @NielsRogge, @ydshieh

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]

File: src/transformers/models/vilt/modeling_vilt.py [hunk 1]
```diff
@@ -1402,3 +1403,94 @@ def forward(
...
+            image_embeds=image_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        sequence_output = outputs[0]
+
+        sequence_output = self.dropout(sequence_output)
+        logits = self.classifier(sequence_output)
```
Comment (NielsRogge) [inline src/transformers/models/vilt/modeling_vilt.py:1476]: I see that you're taking the final hidden states of all tokens here (text + patches), and then applying a classifier on top. 

However, I've seen other implementations that first take the final hidden states of the text tokens only, after which the classifier is applied. Doesn't that seem more logical?

**Review by NielsRogge — CHANGES_REQUESTED** [review]
Thanks for adding this! I wonder what your use case is? You'd like to apply NER on a piece of text for which you also have visual information?

File: src/transformers/models/vilt/modeling_vilt.py (hunk 1 above)
Comment (gilad19) [reply src/transformers/models/vilt/modeling_vilt.py]: I agree, thanks! I modified it accordingly.

**Comment by gilad19** [comment]
Regarding the use case - exactly, apply NER on a piece of text for which you also have a visual information.

**Comment by gilad19** [comment]
Hi @NielsRogge - a gentle reminder :)

File: tests/models/vilt/test_modeling_vilt.py [hunk 2]
```diff
@@ -224,7 +226,7 @@ def _prepare_for_class(self, inputs_dict, model_class, return_labels=False):
                 inputs_dict["labels"] = torch.zeros(
                     self.model_tester.batch_size, self.model_tester.num_labels, device=torch_device
                 )
-            elif model_class.__name__ == "ViltForMaskedLM":
+            elif model_class.__name__ in ["ViltForMaskedLM", "ViltForTokenClassification"]:
```
Comment (NielsRogge) [inline tests/models/vilt/test_modeling_vilt.py:229]: Can you also remove lines 221 and 222 which are commented out?

Also, can you add a `create_and_check_for_token_classification` method (similar to `create_and_check_model`), with a corresponding `test_for_token_classification` method (similar to `test_model`)?

Comment (gilad19) [reply tests/models/vilt/test_modeling_vilt.py:243]: @NielsRogge - Hi, I removed the commented out lines and added the new test test_for_token_classification.

Comment (NielsRogge) [reply tests/models/vilt/test_modeling_vilt.py:243]: Awesome!

File: src/transformers/models/vilt/modeling_vilt.py [hunk 3]
```diff
@@ -1402,3 +1403,92 @@ def forward(
             hidden_states=hidden_states,
             attentions=attentions,
         )
+
+
+@add_start_docstrings(
+    """
+    Vilt Model with a token classification head on top (a linear layer on top of the hidden-states output) e.g. for
```
Comment (NielsRogge) [inline src/transformers/models/vilt/modeling_vilt.py:1410]:
```suggestion
    Vilt Model with a token classification head on top (a linear layer on top of the final hidden-states of the text tokens) e.g. for
```

File: src/transformers/models/vilt/modeling_vilt.py [hunk 4]
```diff
@@ -1402,3 +1403,92 @@ def forward(
...
+
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.num_labels = config.num_labels
+        self.vilt = ViltModel(config, add_pooling_layer=False)
+
+        classifier_dropout = config.hidden_dropout_prob
+
+        self.dropout = nn.Dropout(classifier_dropout)
```
Comment (NielsRogge) [inline src/transformers/models/vilt/modeling_vilt.py:1427]:
```suggestion
        self.dropout = nn.Dropout(config.hidden_dropout_prob)
```

File: src/transformers/models/vilt/modeling_vilt.py [hunk 5]
```diff
@@ -1402,3 +1403,92 @@ def forward(
...
+        head_mask=None,
+        inputs_embeds=None,
+        image_embeds=None,
+        labels=None,
+        output_attentions=None,
+        output_hidden_states=None,
+        return_dict=None,
+    ):
+        r"""
+        labels (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
```
Comment (NielsRogge) [inline src/transformers/models/vilt/modeling_vilt.py:1451]:
```suggestion
        labels (`torch.LongTensor` of shape `(batch_size, text_sequence_length)`, *optional*):
```

**Review by NielsRogge — APPROVED** [review]
Good for me when you apply the suggestions.

**Comment by LysandreJik** [comment]
Feel free to merge when satisfied @NielsRogge

**Pull request merged**